### PR TITLE
AppRootPage: Reduce flickering while loading plugin

### PR DESCRIPTION
**Pull request by torkelo**
A big problem for app plugins (especially the cloud home) is how it temporarily shows a section nav and standard page named "Home" while loading. 

It looks like React 18 or some other changes in main has made this problem slightly less noticeable but think this is a good change anyway 

* Fixes bug if you load an App with a PluginPage layout set to canvas as the first page it would still show left side section nav 
* Default to Canvas layout to reduce flickering when loading an app plugin that uses canvas layout (for example the hg home app) 
* While the plugin load use an empty page title to reduce flickering of plugin name in page header while plugin loads

### CODE REVIEW

File: public/app/core/components/AppChrome/AppChromeService.tsx [hunk 1]
```diff
@@ -77,6 +77,7 @@ export class AppChromeService {
     if (newState.sectionNav !== current.sectionNav || newState.pageNav !== current.pageNav) {
       if (
         newState.actions === current.actions &&
+        newState.layout === current.layout &&
```
Comment (torkelo) [inline public/app/core/components/AppChrome/AppChromeService.tsx:80]: this is the bugfix part that fixes the issue of loading an app plugin page with layout canvas as the first page

**Review by mckn — APPROVED** [review]
Tested it and looks way better! Great work on this!

**Review by ashharrison90 — APPROVED** [review]
lgtm 🙌

**Comment by torkelo** [comment]
@ashharrison90 should we backport this to improve this in 9.5?

**Comment by ashharrison90** [comment]
@torkelo the changes in `AppChromeService` would conflict (since layout doesn't exist in `AppChromeService` in the 9.5.x branch)

if you wanted you could manually backport just the change in `AppRootPage`? it's obviously better, but not exactly critical, so up2u 👍

**Review by leventebalogh — APPROVED** [review]
Nice 👏

**Pull request merged**